R/insertPackage.R

```diff
@@ -159,75 +159,68 @@ identifyPackageType <- function(file) {
     return(ret)
 }
 
+##' This function returns the compile-time information added
+##' to the \code{DESCRIPTION} file in the package.
+##'
+##' @title Get information from a binary package 
+##' @param file the fully qualified path of the package
+##' @return A named vector with several components
+##' @author Dirk Eddelbuettel
+getPackageInfo <- function(file) {
+    if (!file.exists(file)) stop("File ", file, " not found!", call.=FALSE)
+    
+    td <- tempdir()
+    if (grepl(".zip$", file)) {
+        unzip(file, exdir=td)
+    } else if (grepl(".tgz$", file)) {
+        untar(file, exdir=td)
+    } else {
+        ##stop("Not sure we can handle ", file, call.=FALSE)
+        fields <- c("Source"=TRUE, "Rmajor"=NA, "Mavericks"=FALSE)
+        return(fields)
+    }
+
+    pkgname <- gsub("^([a-zA-Z0-9]*)_.*", "\\1", basename(file))
+    path <- file.path(td, pkgname, "DESCRIPTION")
+    builtstring <- read.dcf(path, 'Built')
+    unlink(file.path(td, pkgname), recursive=TRUE)
+
+    fields <- strsplit(builtstring, "; ")[[1]]
+    names(fields) <- c("Rversion", "OSflavour", "Date", "OS")
+
+    rmajor <- gsub("^R (\\d\\.\\d)\\.\\d.*", "\\1", fields["Rversion"])
+    isDarwin13 <- ifelse(fields["OSflavour"] == "x86_64-apple-darwin13.4.0", "yes", "no")
+    fields <- c(fields, "Rmajor"=unname(rmajor), "Mavericks"=unname(isDarwin13))
```

Comment:
> The code path goes over each to-be-inserted file exactly one and inserts it into the computed path.

well, then the architecture stopped working for the changed specs. happens to the best of us :wink: 

> older than Mavericks OS X

won’t that be a problem on the user side instead of the server (drat running) side?

a server may be whatever OS, but a user will use `install.packages` on OS X 10.veryold and wonder why it says that no packages can be found.

or am i missing something?